src/serverenvironment.cpp

```diff
@@ -484,27 +484,25 @@ bool ServerEnvironment::removePlayerFromDatabase(const std::string &name)
 bool ServerEnvironment::line_of_sight(v3f pos1, v3f pos2, float stepsize, v3s16 *p)
 {
 	float distance = pos1.getDistanceFrom(pos2);
-
-	//calculate normalized direction vector
-	v3f normalized_vector = v3f((pos2.X - pos1.X)/distance,
-		(pos2.Y - pos1.Y)/distance,
-		(pos2.Z - pos1.Z)/distance);
-
-	//find out if there's a node on path between pos1 and pos2
-	for (float i = 1; i < distance; i += stepsize) {
-		v3s16 pos = floatToInt(v3f(normalized_vector.X * i,
-			normalized_vector.Y * i,
-			normalized_vector.Z * i) +pos1,BS);
-
-		MapNode n = getMap().getNodeNoEx(pos);
-
-		if(n.param0 != CONTENT_AIR) {
-			if (p) {
-				*p = pos;
-			}
+	// For compatibility, reduce the length of the line
+	// by stepsize, keeping the end's position
+	if (distance < stepsize) {
+		return true;
```

Comment:
couldn't rounding errors mean that two nodes next to each other won't be checked if distance~=stepsize? Also, either of the ends could be a node. I guess this is a logical limitation however

Reply:
`stepsize` is 1 by default. If distance is less than1, eg: the nodes are the same, this will always return true even if the block is solid. Does this match the old behaviour? Is it even good to have the default `stepsize` as 1 if 0 is likely to give a more predicted result